core/src/Kokkos_CopyViews.hpp

```diff
@@ -3709,7 +3709,7 @@ auto create_mirror_view_and_copy(
                 "not explicitly allow padding!");
 
   // same behavior as deep_copy(src, src)
-  if (!alloc_prop_input::has_execution_space)
+  if constexpr (!alloc_prop_input::has_execution_space)
```

Comment:
This looks like an unrelated unnecessary change and it will likely conflict with #6955